File: frontend/src/lib/slopsLimits.ts
export const MAX_FILES_PER_TURN = 5;
export const MAX_SINGLE_FILE = 5 * 1024 * 1024;
export const MAX_TOTAL_UPLOAD = 10 * 1024 * 1024;

export function formatSize(n: number): string {
  if (n < 1024) return `${n} B`;
  if (n < 1024 * 1024) return `${(n / 1024).toFixed(1)} KB`;
  return `${(n / (1024 * 1024)).toFixed(1)} MB`;
}

File: website/slops_limits.py
"""Shared constants for slops file-upload limits. Mirror in frontend/src/lib/slopsLimits.ts."""

MAX_FILES_PER_TURN = 5
MAX_SINGLE_FILE = 5 * 1024 * 1024  # 5 MB
MAX_TOTAL_UPLOAD = 10 * 1024 * 1024  # 10 MB per turn
PREVIEW_MAX_BYTES = 64 * 1024  # 64 KB

# Extensions split into "text/code" (previewable) and "binary" (storage only for now).
TEXT_EXTENSIONS = frozenset(
    {
        ".txt",
        ".md",
        ".json",
        ".yaml",
        ".yml",
        ".csv",
        ".tsv",
        ".log",
        ".py",
        ".js",
        ".ts",
        ".tsx",
        ".jsx",
        ".html",
        ".css",
        ".sh",
        ".toml",
        ".xml",
        ".sql",
    }
)
BINARY_EXTENSIONS = frozenset({".pdf", ".png", ".jpg", ".jpeg", ".gif", ".webp", ".heic", ".docx", ".xlsx", ".pptx"})
ALLOWED_EXTENSIONS = TEXT_EXTENSIONS | BINARY_EXTENSIONS
